sql_grader/problem.py

```diff
@@ -93,25 +111,41 @@ def clone_database(source):
         return destination
 
     @classmethod
-    def run_query(cls, source, query, verify_query=None):
+    def run_query(cls, source, query, verify_query=None,
+                  modification_query=None):
         """
-        Execute the provided SQL query against a copy of the database
+        Execute the provided answer and verification queries against a copy of
+        the database
         """
-        def run(database, query):
+        def run(database, query, is_single_query):
             result = []
             message = None
             with database as connection:
                 try:
-                    for row in connection.execute(query):
+                    if is_single_query:
+                        executor_func = connection.execute
+                    else:
+                        executor_func = connection.executescript
+                    rows = executor_func(query)
+                    for row in rows:
                         result.append(row)
                 except Exception as error:  # pylint: disable=broad-except
                     result = None
                     message = str(error)
             return result, message
+
         database = cls.clone_database(source)
-        result, error = run(database, query)
+
+        if verify_query:
+            is_single_query = False
+        else:
+            is_single_query = True
+        result, error = run(database, query, is_single_query)
+
         if verify_query:
-            result, _ = run(database, verify_query)
+            if modification_query:
+                result, _ = run(database, modification_query, False)
+            result, error = run(database, verify_query, True)
```

Comment:
The first two parameters are read very well (_run… in the DB… a query…_), but to understand the 3rd one (_True…_), we must look at the function definition. Could you use `is_single_query=True` to make it more readable? (and `=False` above)